src/app/rpmostree-builtin-rebase.c

```diff
@@ -74,21 +79,50 @@ rpmostree_builtin_rebase (int             argc,
                                        error))
     return EXIT_FAILURE;
 
-  if (argc < 2 || argc > 3)
+  if (argc > 3)
     {
-      rpmostree_usage_error (context, "Too few or too many arguments", error);
+      rpmostree_usage_error (context, "Too many arguments", error);
       return EXIT_FAILURE;
     }
 
-  new_provided_refspec = argv[1];
-
-  if (argc == 3)
-    revision = argv[2];
-
   if (!rpmostree_load_os_proxy (sysroot_proxy, opt_osname,
                                 cancellable, &os_proxy, error))
     return EXIT_FAILURE;
 
+  if (argc < 2 && !(opt_branch || opt_remote))
+    {
+      return rpmostree_usage_error (context, "Must specify refspec, or -b branch or -r remote", error), EXIT_FAILURE;
+    }
+  else if (argc >= 2)
+    {
+      new_provided_refspec = argv[1];
+      if (argc == 3)
+        revision = argv[2];
+    }
+  else
+    {
+      GVariant *default_deploy = rpmostree_os_get_default_deployment (os_proxy);
+      g_assert (default_deploy);
+      g_autoptr(GVariantDict) dict = g_variant_dict_new (default_deploy);
+      const char *origin_refspec;
+      g_autofree char *remote = NULL;
+      g_autofree char *branch = NULL;
+
+      if (!g_variant_dict_lookup (dict, "origin", "&s", &origin_refspec))
+        return glnx_throw (error, "No origin"), EXIT_FAILURE;
+
+      if (!ostree_parse_refspec (origin_refspec, &remote, &branch, error))
+        return EXIT_FAILURE;
+
+      /* Allow using -m '' */
+      const char *target_remote = (opt_remote && *opt_remote) ?: remote;
```

Comment:
Hmm, would it make more sense to make `-m ''` mean we're rebasing to a local branch? (Otherwise, I don't think that's possible at all if you're not already on a local branch, it seems like.) They already get the "re-use same remote" behaviour by just not specifying `-m` at all, right? I.e., changing this to a simple:

```c
const char *target_remote = opt_remote ?: remote;
```